src/Field/Email.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form\Field;
+
+use InvalidArgumentException;
+use Yiisoft\Form\Field\Base\EnrichmentFromRules\EnrichmentFromRulesInterface;
+use Yiisoft\Form\Field\Base\EnrichmentFromRules\EnrichmentFromRulesTrait;
+use Yiisoft\Form\Field\Base\InputField;
+use Yiisoft\Form\Field\Base\Placeholder\PlaceholderInterface;
+use Yiisoft\Form\Field\Base\Placeholder\PlaceholderTrait;
+use Yiisoft\Form\Field\Base\ValidationClass\ValidationClassInterface;
+use Yiisoft\Form\Field\Base\ValidationClass\ValidationClassTrait;
+use Yiisoft\Html\Html;
+use Yiisoft\Validator\Rule\HasLength;
+use Yiisoft\Validator\Rule\Regex;
+use Yiisoft\Validator\Rule\Required;
+
+use function is_string;
+
+/**
+ * @link https://html.spec.whatwg.org/multipage/input.html#email-state-(type=email)
+ * @link https://developer.mozilla.org/docs/Web/HTML/Element/input/email
+ */
+final class Email extends InputField implements EnrichmentFromRulesInterface, PlaceholderInterface, ValidationClassInterface
+{
+    use EnrichmentFromRulesTrait;
+    use PlaceholderTrait;
+    use ValidationClassTrait;
+
+    /**
+     * Maximum length of value.
+     *
+     * @param int|null $value A limit on the number of characters a user can input.
+     *
+     * @link https://html.spec.whatwg.org/multipage/input.html#attr-input-maxlength
+     * @link https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-maxlength
+     */
+    public function maxlength(?int $value): self
+    {
+        $new = clone $this;
+        $new->inputAttributes['maxlength'] = $value;
+        return $new;
+    }
+
+    /**
+     * Minimum length of value.
+     *
+     * @param int|null $value A lower bound on the number of characters a user can input.
+     *
+     * @link https://html.spec.whatwg.org/multipage/input.html#attr-input-minlength
+     * @link https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-minlength
+     */
+    public function minlength(?int $value): self
+    {
+        $new = clone $this;
+        $new->inputAttributes['minlength'] = $value;
+        return $new;
+    }
+
+    /**
+     * Allow to specify more than one value.
+     *
+     * @param bool $value Whether the user is to be allowed to specify more than one value.
+     *
+     * @link https://html.spec.whatwg.org/multipage/input.html#attr-input-multiple
+     */
+    public function multiple(bool $value = true): self
```

Comment:
For boolean fields let's ensure there's methods like this accepting value and no methods like `withoutX()`. I've seen these in other classes (do search "on without" in IDE please).

Reply:
I don't found methods `without*()`